### ISSUE TITLE
Scaffolded content

### ISSUE THREAD

**Issue by yishaigalatzer** [NuGet/Home#1792]
We want to support adding file to a project (with the intent of them being modifiable by the user).

This is the second part to the content in transitive restore discussion started here - 
https://github.com/NuGet/Home/issues/627 and https://github.com/NuGet/Home/wiki/%5BSpec%5D-Content-v2-for-project.json

The goal is to enable authoring packages that allow for modifying the project in the following scenarios
1. Add a configuration file
2. Add a configuration/code file that is modifiable based on the project 
3. Run XDT transforms over the project

**Comment by 304NotModified** [comment]
We really like to see this feature for NLog.

**Comment by nbarbettini** [comment]
Agreed, this is a current pain point with transitive restore - which is otherwise awesome. :smile:

**Comment by v2Shridhar** [comment]
Any updates, when this feature will be released?

**Comment by lordofscripts** [comment]
Another use case is when you have a series of Views and even a controller in  a web application that you would like to reuse in other web applications. Currently you can't put views in a NUPKG (at least not since I started .NET COre) and you want the user to be able to customize that controller and views to something that blends more with their web application or do some extra things.

I used to be able to do that with standard ASP.NET MVC but now I cannot with ASP.NET Core and it is a bummer. I had to create a NuGet Tripe (as I call it) to do it until this feature is brought back.

**Comment by brgrz** [comment]
It's unclear from the docs and the two GitHub issues whether packaging content is still supported or not.

This page says it is not (or is vague about it)
https://docs.microsoft.com/en-us/nuget/create-packages/project-json-impact

> Similar to install scripts, transforms run on package install and are typically not idempotent. Since there is no install time anymore, XDT Transform and similar features are not supported, and are ignored if such a package is used in a transitive scenario.

> Support for content is currently disabled for similar reasons for scripts and transforms, but we are in the process of designing support for content.

> Content files can still be carried inside the packages, and are ignored currently, however the end user can still copy them into the right spot.

Having hard time understanding what it means "copy them into the right spot" - are we supposed to manually extract and copy content files?

Also, another page at docs says transforms are still supported and gives a whole tutorial how to do it:
https://docs.microsoft.com/en-us/nuget/create-packages/source-and-config-file-transformations

It'd be nice if some cleared up what's the actual status of this and also revised the docs.

**Comment by Jtango18** [comment]
I just want to clarify here (having spent several hours messing with this) that currently, with a PackageReference formatted package, it is **NOT** possible to add a transformed source file (e.g. _MyClass.cs.pp_) to the consuming project's structure in a way that is user visible/editable. Is that correct?

**Comment by chuanboz** [comment]
Any update for this? I see it labled with Priority 1 but has a conflict milestones as Backlog.
